docs/nada-lang-types.md

```diff
@@ -0,0 +1,52 @@
+# Data Types
+
+Overview of the primitive and the compound Nada data types.
+
+## Primitive Data Types
+
+| Public          | Secret                |
+| --------------- | --------------------- |
+| `Integer`         | `SecretInteger`         |
+| `UnsignedInteger` | `SecretUnsignedInteger` |
+| -                | `SecretBlob`            |
+
+## Compound Data Types *(Experimental Feature)*
+
+:::warning
+
+Compound types are a new feature in the Nada language. We hope you find them handy, though there might still be undiscovered bugs. If you encounter anything, please flag them by opening an issue in [nillion-python-starter](https://github.com/nillion-oss/nillion-python-starter).
+
+:::
+
+Nada also support compound types, which include `Array`s and `Tuple`s.
+
+A `Tuple` has two fields, which can be any combination of a [Primitive Data Type](/nada-lang-types#primitive-data-types) or a [Compound Data Type](/nada-lang-types#compound-data-types-experimental-feature). For example a `Tuple` can be as simple as:
+```python
+a = SecretInteger(Input(name="a", party=party1))
+b = SecretInteger(Input(name="b", party=party1))
+my_tuple = Tuple.new(a, b)
+```
+or more complicated like:
+```python
+a = PublicInteger(Input(name="a", party=party1))
+b = SecretInteger(Input(name="b", party=party1))
+c = PublicInteger(Input(name="c", party=party1))
+
+my_tuple = Tuple.new(a, Tuple.new(b, c))
+```
+
+An `Array` on the other hand has a fixed number of fields which can be any of a [Primitive Data Type](/nada-lang-types#primitive-data-types) or a [Nada Compound Data Type](/nada-lang-types#compound-data-types-experimental-feature), *but all need to be of the same type*. For example:
```

Comment:
Maybe, I would move arrays before tuples for following the order you presented them above.

Besides, Arrays have another constructor:

`a = Array(PublicInteger(Input(name="c", party=party1)), size=3)`

Finally, we can combine them, maybe we should comment on it.